### PR TITLE
NEP: Expansion of gufunc signatures.

### PR DESCRIPTION
**Pull request by mhvk**
As requested on the mailing list.

### CODE REVIEW

File: doc/neps/nep-0020-gufunc-signature-enhancement.rst [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+  <Dimension name>       ::= valid Python variable name | valid integer
+  <Dimension modifier>   ::= nil | "|1" | "?"
+
+#. All quotes are for clarity.
+#. Unmodified core dimensions that share the same name must have the same size.
+   Each dimension name typically corresponds to one level of looping in the
+   elementary function's implementation.
+#. White spaces are ignored.
+#. An integer as a dimension name freezes that dimension to the value.
+#. If a name if suffixed with the "|1" modifier, it is allowed to broadcast
```
Comment (mattip) [inline doc/neps/nep-0020-gufunc-signature-enhancement.rst]: build is failing, should this be `` not "? Try to do `make html` in `doc/neps` (using python 3.6 for f-strings)

Comment (mhvk) [reply doc/neps/nep-0020-gufunc-signature-enhancement.rst]: Thanks, that was indeed the problem. Plus a few others that are now fixed...

File: doc/neps/nep-0020-gufunc-signature-enhancement.rst [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+1. Fixed-size dimensions.  Code working with spatial vectors often explicitly
+   is for 2 or 3-dimensional space (e.g., the code from the `Standards Of
+   Fundamental Astronomy <http://www.iausofa.org/>`_, which the author hopes
+   to wrap using gufuncs for astropy [2]_).  The signature should be able to
+   indicate that.  E.g., the signature of a function that converts a polar
+   angle to a two-dimensional cartesian unit vector would currently have to be
+   ``()->(n)``, with there being no way to indicate that ``n`` has to equal 2.
+   Indeed, this signature is particularly annoying since without putting in an
+   output argument, the current gufunc wrapper code fails because it cannot
+   deteermine ``n``.  Similarly, the signature for an cross product of two
```
[the remark this answers is not in the document]
Comment (mhvk) [reply doc/neps/nep-0020-gufunc-signature-enhancement.rst]: thanks, corrected.

**Comment by mattip** [comment]
It may be worth mentioning somewhere that we could accept 1,2, or all three of the extensions.

**Comment by mhvk** [comment]
@mattip - I sent a note to the mailing list which explicitly mentions this.

**Pull request merged**